proxy/http/HttpBodyFactory.cc

```diff
@@ -196,44 +196,24 @@ HttpBodyFactory::fabricate_with_old_api(const char *type, HttpTransact::State *c
 void
 HttpBodyFactory::dump_template_tables(FILE *fp)
 {
-  RawHashTable *h1, *h2;
-  RawHashTable_Key k1, k2;
-  RawHashTable_Value v1, v2;
-  RawHashTable_Binding *b1, *b2;
-  RawHashTable_IteratorState i1, i2;
-  HttpBodySet *body_set;
-
   lock();
-
-  h1 = table_of_sets;
-
-  if (h1 != nullptr) {
-    ///////////////////////////////////////////
-    // loop over set->body-types hash table //
-    ///////////////////////////////////////////
-
-    for (b1 = h1->firstBinding(&i1); b1 != nullptr; b1 = h1->nextBinding(&i1)) {
-      k1       = table_of_sets->getKeyFromBinding(b1);
-      v1       = table_of_sets->getValueFromBinding(b1);
-      body_set = (HttpBodySet *)v1;
-
-      if (body_set != nullptr) {
-        fprintf(fp, "set %s: name '%s', lang '%s', charset '%s'\n", k1, body_set->set_name, body_set->content_language,
-                body_set->content_charset);
+  if (table_of_sets) {
+    for (auto it1 = table_of_sets->begin(); it1 != table_of_sets->end(); ++it1) {
+      HttpBodySet *body_set = (HttpBodySet *)it1->second;
```

Comment:
No C-style casts! Why does this need casting at all?

Reply:
the value type is `HttpBodySetRawData`, the parent class of `HttpBodySet `.

Reply:
You can still use a c++ cast

Reply:
Use `static_cast` in that case.